src/app/(projects)/my-projects/page.tsx

```diff
@@ -9,7 +10,11 @@ import {prefetchGetMyContestsReportCounts} from '@/lib/queries/contest/getMyCont
 import MyContests from '@/components/sections/projects/MyContests'
 
 const MyProjectsPage = async () => {
-  await requirePageSession()
+  const session = await requirePageSession()
+
+  if (!session.user.role) {
+    redirect('/confirm-path')
```

Comment:
Not sure about this design, I think this should be handled during the sign in, otherwise we would have to add this to every page, that user can be redirected to after login. Also, in `signIn` functions in `OAuthButtons`, we'll have at least 3 different pages where we want to navigate next (POs dashboard - My projects, Auditors dashboard, Judges dashboard), not just My projects.

One solution that comes to my mind would be as follows:
- sign in page accepts redirectUrl (next page) as search param
- in `OAuthButtons`, use Confirm path page for signIn callbackUrl and append next page as search param
- on Confirm path page, fetch user role from DB
- if role is present, redirect immediately to the next page (taking the next page from search params), otherwise let user select his role

Alternatively, instead of redirecting to Confirm path page everytime, maybe we could use [redirect: false](https://next-auth.js.org/getting-started/client#using-the-redirect-false-option) option and fetch user data directly on sign in page after the login and redirect user to the next page accordingly (i.e. either to redirectUrl or to Confirm path page with redirectUrl as search param). This would probably be even cleaner way🤷

---
Secondly, there is a use case, when user who is not logged in, tries to access a protected page - currently he is redirected to Next auth sign in page with SessionRequired error and a callbackUrl containing next page (check requirePageSession). This should also be handled by the new login page, e.g. get error from search params, display some message and use the next page URL as callback URL in `signIn`.

If you want, we can also have a call tomorrow to discuss it in more detail, what would be the best approach.

Reply:
Great points. I like the first solution better as it utilizes next server components more and minimizes the client side logic. It's still a pretty clean solution as the `callbackUrls` are just passed around to the different pages. This also integrates nicely with the `SessionRequired` use case you pointed out 👍

Reply:
```ts
if (!session.user.role) {
    redirect('/confirm-path')
}
```

Also, I think we still need to keep this condition in the `my-projects` page and in subsequent dashboard pages, as there is a use case when a user signs in, doesn't select a role and goes to the home page. When the user goes to the dashboard there would be no prompt to select a role.

Reply:
Came up with a solution to check for the user role in `requirePageSession` and redirect to the `select-role` page if the role is not present. This way we don't have to add this check to every page needed.